packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -60,24 +60,134 @@ class StorageReference {
         <String, String>{'path': _pathComponents.join("/")});
   }
 
+  /// Retrieves metadata associated with an object at this [StorageReference].
+  Future<StorageMetadata> getMetadata() async {
+    return new StorageMetadata._fromMap(await FirebaseStorage._channel
+        .invokeMethod("StorageReference#getMetadata", <String, String>{
+      'path': _pathComponents.join("/"),
+    }));
+  }
+
   String get path => _pathComponents.join('/');
 }
 
+/// Metadata for a [StorageReference]. Metadata stores default attributes such as
+/// size and content type.
+class StorageMetadata {
+  StorageMetadata({
+    this.cacheControl,
+    this.contentDisposition,
+    this.contentEncoding,
+    this.contentLanguage,
+    this.contentType,
+  }) : bucket = null,
+       generation = null,
+       metadataGeneration = null,
+       path = null,
+       name = null,
+       sizeBytes = null,
+       creationTimeMillis = null,
+       updatedTimeMillis = null,
+       md5Hash = null;
+
+  StorageMetadata._fromMap(Map<String, dynamic> map)
```

Comment:
The argument needs to be `Map<dynamic, dynamic>` because that is what you get on the channel.